### ISSUE TITLE
Break build after install

### ISSUE THREAD

**Issue by julienfollow** [FusionAuth/nodebb-plugin-fusionauth-oidc#12]
HEllo
When i install ur plugin in my nodebb system, build works but not start. 
Just ur plugin the issue. 

Any solutions ? 

Thanks
```
2021-12-22T17:51:34.715Z [4567/115] - error: uncaughtException: Cannot convert undefined or null to object
TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at trim (/usr/src/app/src/settings.js:27:35)
    at trim (/usr/src/app/src/settings.js:31:4)
    at mergeSettings (/usr/src/app/src/settings.js:44:3)
    at Settings.checkStructure (/usr/src/app/src/settings.js:233:3)
    at Settings.<anonymous> (/usr/src/app/src/settings.js:69:9)
    at /usr/src/app/src/settings.js:103:13
    at /usr/src/app/src/promisify.js:43:50
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {"date":"Wed Dec 22 2021 17:51:34 GMT+0000 (Coordinated Universal Time)","error":{},"exception":true,"os":{"loadavg":[5.28,4.76,4.18],"uptime":121397.97},"process":{"argv":["/usr/local/bin/node","/usr/src/app/app.js"],"cwd":"/usr/src/app","execPath":"/usr/local/bin/node","gid":1000,"memoryUsage":{"arrayBuffers":2765213,"external":5411772,"heapTotal":100519936,"heapUsed":78528896,"rss":155144192},"pid":115,"uid":1000,"version":"v16.13.1"},"stack":"TypeError: Cannot convert undefined or null to object\n    at Function.entries (<anonymous>)\n    at trim (/usr/src/app/src/settings.js:27:35)\n    at trim (/usr/src/app/src/settings.js:31:4)\n    at mergeSettings (/usr/src/app/src/settings.js:44:3)\n    at Settings.checkStructure (/usr/src/app/src/settings.js:233:3)\n    at Settings.<anonymous> (/usr/src/app/src/settings.js:69:9)\n    at /usr/src/app/src/settings.js:103:13\n    at /usr/src/app/src/promisify.js:43:50\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","trace":[{"column":null,"file":null,"function":"Function.entries","line":null,"method":"entries","native":false},{"column":35,"file":"/usr/src/app/src/settings.js","function":"trim","line":27,"method":null,"native":false},{"column":4,"file":"/usr/src/app/src/settings.js","function":"trim","line":31,"method":null,"native":false},{"column":3,"file":"/usr/src/app/src/settings.js","function":"mergeSettings","line":44,"method":null,"native":false},{"column":3,"file":"/usr/src/app/src/settings.js","function":"Settings.checkStructure","line":233,"method":"checkStructure","native":false},{"column":9,"file":"/usr/src/app/src/settings.js","function":null,"line":69,"method":null,"native":false},{"column":13,"file":"/usr/src/app/src/settings.js","function":null,"line":103,"method":null,"native":false},{"column":50,"file":"/usr/src/app/src/promisify.js","function":null,"line":43,"method":null,"native":false},{"column":21,"file":"node:internal/process/task_queues","function":"processTicksAndRejections","line":83,"method":null,"native":false}]}
2021-12-22T17:51:34.716Z [4567/115] - error: TypeError: Cannot convert undefined or null to object
```

**Comment by joolsr** [comment]
I get same issue with v1.19.1

**Comment by zhivkoangelov** [comment]
I got the same error. 

Additional details:
2022-03-29T14:55:35.189Z [5555/22844] - warn: [plugins] Unable to load library for: nodebb-plugin-fusionauth-oidc
2022-03-29T14:55:35.190Z [5555/22844] - error: ReferenceError: settings is not defined
    at Object.<anonymous> (C:\zan\repos\vial_repos\vial-forum-nodebb\src\settings.js:88:24)

Did anyone solved this issue ?

**Comment by jimmiejackson414** [comment]
Doesn't look like it's been addressed or this project has been abandoned. I just tried installing and am receiving the same error

**Comment by mooreds** [comment]
Thanks for the report and sorry about this!

I know we're running it for our community forum, so maybe we made changes that haven't been upstreamed. I'll take a look.

**Comment by mooreds** [comment]
No changes I can see. It looks like we are running version 1.0.1 which is the latest version in the package.json. We are running v 1.18.6.

Is everyone here running this plugin against FusionAuth or are you running against different OIDC servers?

**Comment by jimmiejackson414** [comment]
@mooreds I'd followed this [FusionAuth article](https://fusionauth.io/blog/2020/05/13/setting-up-single-sign-on-for-nodebb) yesterday, using a fresh NodeBB install from two days prior.

**Comment by mooreds** [comment]
@jimmiejackson414 what version of nodebb are you running?

**Comment by jimmiejackson414** [comment]
@mooreds v1.18.6

**Comment by mooreds** [comment]
I just merged https://github.com/FusionAuth/nodebb-plugin-fusionauth-oidc/pull/7 and made a new release. Can you please try that? I am also opening a ticket with our nodebb provider to see if they have done anything different to the plugin; will report back.

**Comment by jimmiejackson414** [comment]
@mooreds quick question: I've been reinstalling fresh builds to test different things, and the latest install is nodeBB v2.0.1. Do you know if this plugins works with that version as well? Otherwise I'll go ahead and downgrade and test this plugin.

**Comment by jimmiejackson414** [comment]
@mooreds When I search for the plugin, I'm still only seeing v1.0.1

![Screen Shot 2022-06-01 at 11 00 36 AM](https://user-images.githubusercontent.com/93394140/171471939-db14f922-d77f-4637-9b93-1b9345bcbf11.png)

**Comment by mooreds** [comment]
Ah, my mistake. Please try now and grab 1.0.3. (That included a small upgrade to typescript required for publishing.)

**Comment by jimmiejackson414** [comment]
Odd, I'm having issues opening localhost:9011 following that FusionAuth guide I posted earlier. I'm using v.18.6 of NodeBB, but that makes me think that it's not really an issue with this specific plugin (I haven't installed it yet).

**Comment by jimmiejackson414** [comment]
@mooreds Finally got past that issue. Now I'm not seeing the option for "OpenID Connect" in the Plugins dropdown? I've already rebuilt and restarted nodebb, and performed several page refreshes. Is this an issue with the plugin?

**Comment by jimmiejackson414** [comment]
Just a follow up question, do you know if this plugin happens to work with v2.x of NodeBB?

**Comment by mooreds** [comment]
@jimmiejackson414 does the error go away when you delete/disable this plugin?

**Comment by mooreds** [comment]
Okay. I need to set it up and debug it myself. That'll probably be a week or so from now, just because of my current schedule.

**Comment by jimmiejackson414** [comment]
Great, thank you for the update!

**Comment by Rnaaz** [comment]
Hello,
I am also facing the same issue. Looks like the plugin has some issue. Even I upgraded the nodebb to version 1.19.8 and the fusionauth plugin I am using shows version 1.0.1.  Any ETA for this issue?

**Comment by ThorodanBrom** [comment]
Hello @mooreds, I've pushed a PR for the fix - it works when I tested it on NodeBB v1.19.1.

**Comment by Rnaaz** [comment]
> Hello @mooreds, I've pushed a [PR](https://github.com/FusionAuth/nodebb-plugin-fusionauth-oidc/pull/14) for the fix - it works when I tested it on NodeBB v1.19.1.

**Comment by Rnaaz** [comment]
We have also done the same and it works fine. Though We don't know if this is the correct approach. This [PR](https://github.com/FusionAuth/nodebb-plugin-fusionauth-oidc/pull/14) works.

**Comment by mooreds** [comment]
Thanks everyone, I just merged #14 . Thanks so much to @ThorodanBrom for submitting it. I'll push a release today.

Am going to close this issue out when that is done.

**Comment by mooreds** [comment]
Fixed in 1.0.4
